### PR TITLE
executor: support show table partition regions

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

related parser PR https://github.com/pingcap/parser/pull/855

As the title said, here is an example: 

```sql
mysql>show table t partition (p4) regions;
+-----------+----------------+----------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| REGION_ID | START_KEY      | END_KEY        | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
+-----------+----------------+----------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| 1047      | t_154_r        | t_154_r_200000 | 1048      | 1               | 1048  | 0          | 0             | 0          | 1                    | 1                |
| 1049      | t_154_r_200000 | t_154_r_400000 | 1050      | 1               | 1050  | 0          | 0             | 0          | 1                    | 0                |
| 1051      | t_154_r_400000 | t_154_r_600000 | 1052      | 1               | 1052  | 0          | 0             | 0          | 1                    | 0                |
| 1053      | t_154_r_600000 | t_154_r_800000 | 1054      | 1               | 1054  | 0          | 0             | 0          | 1                    | 0                |
| 2         | t_154_r_800000 |                | 3         | 1               | 3     | 0          | 747           | 0          | 1                    | 0                |
| 1045      | t_154_         | t_154_r        | 1046      | 1               | 1046  | 0          | 0             | 0          | 1                    | 1                |
+-----------+----------------+----------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
6 rows in set
Time: 0.009s
mysql>show table t partition (p3) regions;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| 1041      | t_153_    | t_154_  | 1042      | 1               | 1042  | 0          | 0             | 0          | 1                    | 0                |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
1 row in set
```
 
### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- 

### Release note 

- support show table partition regions.

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -1488,48 +1510,28 @@ func (e *ShowExec) fetchShowTableRegions() error {
...
 		}
-		regions = append(regions, partitionRegions...)
+		regions = append(regions, rs...)
 	}
 	return regions, nil
 }
 
-func getPartitionIndexRegions(info *model.PartitionInfo, tbl table.PartitionedTable, indexInfo *model.IndexInfo, tikvStore tikv.Storage, splitStore kv.SplittableStore) ([]regionMeta, error) {
-	var regions []regionMeta
+func getTableIndexRegions(physicalIDs []int64, indexInfo *model.IndexInfo, tikvStore tikv.Storage, splitStore kv.SplittableStore) ([]regionMeta, error) {
```
Comment (djshow832) [inline executor/show.go:1526]:
```suggestion
func getTableIndexRegions(indexInfo *model.IndexInfo, physicalIDs []int64, tikvStore tikv.Storage, splitStore kv.SplittableStore) ([]regionMeta, error) {
```
It's weird that the parameter order is different with `getTableRegions`.

File: executor/executor_test.go [hunk 2]
```diff
@@ -4418,16 +4418,43 @@ func (s *testSplitTable) TestShowTableRegion(c *C) {
...
+		re = tk.MustQuery(fmt.Sprintf("show table t partition (p%v) regions", i))
+		rows = re.Rows()
+		c.Assert(len(rows), Equals, 4)
+		p := tbl.Meta().GetPartitionInfo().Definitions[i]
+		c.Assert(rows[0][1], Equals, fmt.Sprintf("t_%d_", p.ID))
+		c.Assert(rows[1][1], Equals, fmt.Sprintf("t_%d_r_1000000", p.ID))
+		c.Assert(rows[2][1], Equals, fmt.Sprintf("t_%d_r_2000000", p.ID))
+		c.Assert(rows[3][1], Equals, fmt.Sprintf("t_%d_r_3000000", p.ID))
+	}
+	re = tk.MustQuery("show table t partition (p0, p4) regions")
```
Comment (djshow832) [inline executor/executor_test.go:4440]: Is it allowed to query `show table t partition (p0, p0) regions`?

Comment (djshow832) [reply executor/executor_test.go:4440]: And what about `show table t partition (p10) regions`?

File: executor/show.go (hunk 1 above)
Comment (crazycs520) [reply executor/show.go:1526]: done.

File: executor/executor_test.go (hunk 2 above)
Comment (crazycs520) [reply executor/executor_test.go:4440]: This pr is used to support the syntax like `show table t partition (p0, p0) regions`.
Already add test for unknown partitions.

Comment (djshow832) [reply executor/executor_test.go:4440]: `(p0, p0)`, I mean, the partitions are duplicate.

Comment (crazycs520) [reply executor/executor_test.go:4440]: great, it's ok,  already add test for test.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**